homeassistant/components/ps4/config_flow.py

```diff
@@ -0,0 +1,124 @@
+"""Config Flow for PlayStation 4."""
+from collections import OrderedDict
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.ps4.const import (
+    DEFAULT_NAME, DEFAULT_REGION, DOMAIN, REGIONS)
+from homeassistant.const import (
+    CONF_CODE, CONF_HOST, CONF_IP_ADDRESS, CONF_NAME, CONF_REGION, CONF_TOKEN)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class PlayStation4FlowHandler(config_entries.ConfigFlow):
+    """Handle a PlayStation 4 config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize the config flow."""
+        self.creds = None
+        self.name = None
+        self.host = None
+        self.region = None
+        self.pin = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle a user config flow."""
+        from pyps4_homeassistant import Helper
+        helper = Helper()
+
+        # Abort if device is configured.
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason='devices_configured')
+
+        # Check if able to bind to ports: UDP 987, TCP 997.
```

Comment:
Is this for the app or would this be needed in after a restart with a configured entry as well? It is not clear why you bind the ports, improve documentation?

Reply:
I bind the ports just to check if the Home Assistant instance. If you cannot bind to the ports then there is no point in continuing. The most likely reason this would fail is if the python runtime does not have access to privileged ports and would need extra configuration described in my docs PR.